Tests/RevenueCatUITests/Templates/ExternalPurchaseAndRestoreTests.swift

```diff
@@ -0,0 +1,391 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  ExternalPurchaseAndRestoreTests.swift
+//
+//  Created by James Borthwick on 2024-06-20.
+
+import Foundation
+import Nimble
+import RevenueCat
+@testable import RevenueCatUI
+import SwiftUI
+import XCTest
+
+#if !os(watchOS) && !os(macOS)
+
+enum TestError: Error {
+    case error
+}
+
+// swiftlint:disable type_body_length
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@MainActor
+class ZZExternalPurchaseAndRestoreTests: TestCase {
```

Comment:
When this class is named `ExternalPurchaseAndRestoreTests`, the test `testSamplePaywallWithLoadingEligibility` in `Template1ViewTests` reliably fails (although it passes when run on my machine).

Reply:
I would maybe add this in a comment in this file until we figure out _exactly_ what is happening 🤷‍♂️

Reply:
LOL that must have been a fun conclusion to reach. I would have thrown my computer out the window.

Reply:
😢 Ya it was quite annoying!